src/main/java/com/example/CleanFolders.java

```diff
@@ -0,0 +1,127 @@
+package com.example;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Egy könyvtárstruktúrában dokumentumokat szerkesztenek.
+ *
+ * Minden dokumentumszerkesztéskor .bak kiterjesztésű fájlok biztonsági mentés jönnek létre
+ * (doc1.doc-hoz doc1.bak), azonban gyakori, hogy a szerkesztő bezárása után ottmaradnak.
+ *
+ * Sok esetben a dokumentumokat már törlik is, és üres könyvtárra egyébként nem lenne szükség,
+ * de a .bak fájlok miatt a könyvtárak megmaradnak.
+ *
+ * A feladat egy olyan program készítése, ami a paraméterként kapott könyvtárból kiindulva
+ * kitisztítja a dokumentum nélkül ottmaradt .bak fájlokat, és az üres könyvtárakat is törli.
+ *
+ * A struktúra egy nagyon nagy (több milliós) virtuális fájlrendszer, és egy törlés viszonylag lassú művelet,
+ * ezért fontos több szálon végezni a törléseket - a fájlrendszer elosztott,
+ * ezért jól fogja bírni a párhuzamos terhelést.
+ *
+ * Megoldásként Java nyelvű forráskódot várunk (tetszőleges Java verzió használható).
+ */
+public class CleanFolders {
+	private static final List<File> deletableBaks = new ArrayList<>();
+	private static final List<File> folders = new ArrayList<>();
+
+	/**
+	 * Detects directories under the given directory and collects all .bak files without corresponding .doc files next to them.
+	 * The method uses checkFolder() for detection and loneBakDelete() for deletion.
+	 * @param startFolder	File of the starting directory.
+	 */
+	public static void bakDeletion(File startFolder) {
+		checkFolder(startFolder);
+		loneBakDelete();
+	}
+
+	/**
+	 * Returns the collected, deletable .bak files.
+	 * Created for opening a safe backdoor for testing purposes.
+	 * @return	List<File> of .bak files.
+	 */
+	public static List<File> getDeletableBaks() {
+		return deletableBaks;
+	}
+
+	/**
+	 * Returns the collected directories.
+	 * Created for opening a safe backdoor for testing purposes.
+	 * @return	List<File> of directories.
+	 */
+	public static List<File> getFolders() {
+		return folders;
+	}
+
+	/**
+	 * Helper method for directory stream filtering and collecting.
```

Comment:
Come on! The comment is longer than the code. It's faster to read the method than the comment and the comment gives zero extra information beyond what I can already see from the code.

Reply:
That's a good thing to discuss again. Universally. :)

In case you have to document what the method does and what is the purpose, but you create a dense code or the code is verbal enough (e.g. Stream chained statements or short- and verbal-named methods), can be interesting to look at it when you are a developer and has the right or time to look into the class code.

Based upon my poor experience on Java, as a 1st case, I would say if someone will use the class, will look on Javadocs hints better then opening the class and checking the code manually. It goes of cource, for visible methods only, not for privates.

Furthermore, as a 2nd case, I understand there is a way to generate a Javadocs hypertext documentation from all Javadocs comments which will not contain the code but the texts themselves.

As 3rd, from my past, I was tought that docs can be longer than code if proper explanation is needed. I know, old-school from the late '90s-'00s :)

What are the present rules and ideas on this topic?